### PR TITLE
add log2(u32) and use it to prevent casting to and from float to int

### PR DESCRIPTION
**Pull request by kevgs**
## PR for #11440 

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11458)

### CODE REVIEW

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -402,7 +406,7 @@ impl WebGLRenderingContext {
         // the returned value of GL_MAX_TEXTURE_SIZE when
         // target is GL_TEXTURE_2D or GL_MAX_CUBE_MAP_TEXTURE_SIZE
         // when target is not GL_TEXTURE_2D.
-        if level > (max as f32).log2() as i32 {
+        if level > log2(max) as i32 {
```
Comment (Manishearth) [inline components/script/dom/webglrenderingcontext.rs]: We no longer need the cast to i32 here, unless max can be zero

Comment (kevgs) [inline components/script/dom/webglrenderingcontext.rs]: level is i32 and log2() is u32. Cast is necessary for compilation.

**Comment by Manishearth** [comment]
@bors r+ 

oh, ok

**Pull request merged**